### DISCUSSION TITLE
enh: customisable model external icon in the selector

### DISCUSSION THREAD

**Discussion by hg0428** [open-webui/open-webui#8766] [Ideas]
If you connect a locally-run OpenAI-compatible API, the models from it show up as external even if they are local. For example, models from LM Studio or Ollama's experimental OpenAI API.
This is one of those small but very annoying problems with Open WebUI. Let's get this fixed.

Also, in set task model, all my local models show up in the external models dropdown.
Perhaps there should be a local->external toggle for each individual connection? Or perhaps custom labels (e.g. Groq, Ollama, my custom API endpoint, LM Studio). I think this later option is the best. Just have the same little link symbol but the text can be customized. No need to change the icon, I think it works for letting you know the source.

![image](https://github.com/user-attachments/assets/5b5595a5-bcc7-4da5-aa56-0aef881d81b8)

**Comment by hg0428** [comment]
Custom labels would also help you distinguish between models if you have the same model from multiple providers. For example, I have Llama3.1 8B locally and Llama3.1 8B via Groq and there is nothing to distinguish them besides manually changing the name.

**Comment by greg-d128** [comment]
I was about to put in similar request. I have two connections to ollama, one local to the laptop and another that points at a box more suitable to running llm. I have multiple models that are very difficult to distinguish.

Ideally though, I would like to put a tag, or an icon on a connection and have all the models from that connection inherit these. That way it will not need to be constantly kept up to date. Even having an option of modifying the model name based on the host it is coming from would help.

Hmm.. I wonder if model files are accessible, maybe a small script that automatically changes model names based on the IP they are coming from?

**Comment by avviano** [comment]
When setting up API (Connections > Setting), use the PREFIX field to distinguish models. So for all my API models from OpenAI I have the PREFIX "OpenAI" set. These will show up (as a prefix) in the dropdown in your chat and help you distinguish between the same models from various providers. Also, you can distinguish APIs from locally hosted models via the "external" symbol (which designates models hosted with external API providers). 

I do have about 40 models where I have difficulty identifying the source (parent) and there is no real way to find out which providers they belong too, as in the individual model settings there is no source information (like which API connection they belong too and so on), which makes things confusing. I wish Open WebUI would provide model source information in the model settings.

The prefix will get you 90% to where you need to be.
